src/org/sosy_lab/java_smt/solvers/cvc5/CVC5InterpolatingProver.java

```diff
@@ -0,0 +1,334 @@
+            .collect(ImmutableSet.toImmutableSet());
+
+    if (formulasOfB.isEmpty()) { // Catch trivial case
+      return mgr.getBooleanFormulaManager().makeBoolean(false);
+    }
+
+    // fit the Input to work with getCVC5Interpolation
+
+    ArrayList<Collection<Term>> formAAsList = new ArrayList<>();
+    ArrayList<Collection<Term>> formBAsList = new ArrayList<>();
+
+    formAAsList.add(formulasOfA);
+    formBAsList.add(formulasOfB);
+
+    ArrayList<ArrayList<Collection<Term>>> interpolPair = new ArrayList<>();
+    interpolPair.add(formAAsList);
+    interpolPair.add(formBAsList);
+
+    Term itp = getCVC5Interpolation(interpolPair);
+
+    BooleanFormula result = creator.encapsulateBoolean(itp);
+
+    return result;
+  }
+
+  @Override
+  public List<BooleanFormula> getSeqInterpolants(
+      List<? extends Collection<Term>> partitionedFormulas)
+      throws SolverException, InterruptedException {
+    Preconditions.checkArgument(
+        !partitionedFormulas.isEmpty(), "at least one partition should be available.");
+
+    final List<BooleanFormula> itps = new ArrayList<>();
+    for (int i = 1; i < partitionedFormulas.size(); i++) {
+      itps.add(
+          getInterpolant(
+              ImmutableList.copyOf(Iterables.concat(partitionedFormulas.subList(0, i)))));
+    }
+    return itps;
+  }
+
+  // Experimental!
+  @Override
+  public List<BooleanFormula> getTreeInterpolants(
+      List<? extends Collection<Term>> pPartitionedFormulas, int[] pStartOfSubTree)
+      throws SolverException, InterruptedException {
+    Preconditions.checkState(!closed);
+    assert InterpolatingProverEnvironment.checkTreeStructure(
+        pPartitionedFormulas.size(), pStartOfSubTree);
+
+    // Generate every Interpolation Pair
+    ArrayList<ArrayList<ArrayList<Collection<Term>>>> interpolationPairs =
+        getTreeInterpolationPairs(pPartitionedFormulas, pStartOfSubTree);
+
+    final ArrayList<Term> itps = new ArrayList<>();
+    try { // Interpolate every Interpolation Pair
+      for (int i = 0; i < interpolationPairs.size(); i++) {
+        itps.add(getCVC5Interpolation(interpolationPairs.get(i)));
+      }
+    } catch (UnsupportedOperationException e) {
+      if (e.getMessage() != null) {
+        throw new SolverException(e.getMessage(), e);
+      } else {
+        throw e;
+      }
+    }
+
+    final List<BooleanFormula> result = new ArrayList<>();
+    for (Term itp : itps) {
+      result.add(creator.encapsulateBoolean(itp));
+    }
+    assert result.size() == pStartOfSubTree.length - 1;
+    return result;
+  }
+
+  /**
+   * Interpolates a Tuple of Interpolants according to Craig-Interpolation using CVC5-Interpolation.
+   *
+   * @param formulaPair Interpolation Pair
+   * @return Interpolation of the Interpolation Pair following the definition of
+   *     Craig-Interpolation.
+   */
+  private Term getCVC5Interpolation(ArrayList<ArrayList<Collection<Term>>> formulaPair) {
```

Comment:
If a parameter is a simple Pair, then please use two separate arguments for the first and second pair-element.

Reply:
Done